.github/workflows/build_and_deploy_infer.yml

```
@@ -18,7 +18,21 @@ defaults:
     working-directory: infer
 
 jobs:
+  validate:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Validate version
+        run: |
+          if [[ "${{ inputs.version }}" =~ ^[0-9]+\.[0-9]+\.[0-9]+(-preview\.[0-9]+)?$ ]]; then
+            echo "Valid version format"
+          else
+            echo "Invalid version format. Expected A.BB.C or A.BB.C-preview.D"
+            exit 1
+          fi
   build:
```

ルールは特に決まってないけどさすがに空けたいかも
```suggestion

  build:
```